tests/integration/utils.py

```diff
@@ -164,6 +144,35 @@ def _get_authentication_params(self):
         return {"auth": None}
 
 
+def download_archive(download_url, archive_path):
+    """
+    Download an archive.
+
+    :param download_url: URL to get the archive
+    :param archive_path: Path to the downloaded bundle
+    """
+    with requests.get(download_url, stream=True) as resp:
+        resp.raise_for_status()
+        with open(archive_path, "wb") as file:
+            for chunk in resp.iter_content(chunk_size=8192):
+                file.write(chunk)
+
+
+def extract_archive(archive_path, source_name):
+    """
+    Extract tar or zip archive.
+
+    :param archive_path: Path to archive to be extracted
+    :param source_name: Path to a directory for extracting data
+    """
+    if tarfile.is_tarfile(archive_path):
+        with tarfile.open(archive_path, "r:gz") as tar_ref:
+            tar_ref.extractall(source_name)
+    elif zipfile.is_zipfile(archive_path):
+        with zipfile.ZipFile(archive_path, "r") as zip_ref:
+            zip_ref.extractall(source_name)
```

Comment:
`shutil` already has a method for this, see https://docs.python.org/3/library/shutil.html#shutil.unpack_archive